### PR TITLE
Fix part of lint/go errors (stylecheck) 

### PR DESCRIPTION
**Pull request by karamaru-alpha**
**What this PR does / why we need it**:

Fix part of errors which `make lint/go` returns (stylecheck).

<details><summary>stylecheck lint errors</summary>

```
$ make lint/go | grep -e "(stylecheck)"
docs/main.go:48:38: ST1013: should use constant http.StatusTemporaryRedirect instead of numeric literal 307 (stylecheck)
pkg/app/piped/trigger/trigger.go:180:3: ST1003: var repoId should be repoID (stylecheck)
pkg/app/server/grpcapi/piped_api.go:45:6: ST1003: type pipedApiApplicationStore should be pipedAPIApplicationStore (stylecheck)
pkg/app/server/grpcapi/piped_api.go:54:6: ST1003: type pipedApiDeploymentStore should be pipedAPIDeploymentStore (stylecheck)
pkg/app/server/grpcapi/piped_api.go:66:6: ST1003: type pipedApiDeploymentChainStore should be pipedAPIDeploymentChainStore (stylecheck)
pkg/app/server/grpcapi/piped_api.go:71:6: ST1003: type pipedApiPipedStore should be pipedAPIPipedStore (stylecheck)
pkg/app/server/grpcapi/piped_api.go:76:6: ST1003: type pipedApiEventStore should be pipedAPIEventStore (stylecheck)
pkg/app/server/grpcapi/web_api.go:51:6: ST1003: type webApiApplicationStore should be webAPIApplicationStore (stylecheck)
pkg/app/server/grpcapi/web_api.go:61:6: ST1003: type webApiDeploymentChainStore should be webAPIDeploymentChainStore (stylecheck)
pkg/app/server/grpcapi/web_api.go:66:6: ST1003: type webApiDeploymentStore should be webAPIDeploymentStore (stylecheck)
pkg/app/server/grpcapi/web_api.go:71:6: ST1003: type webApiPipedStore should be webAPIPipedStore (stylecheck)
pkg/app/server/grpcapi/web_api.go:83:6: ST1003: type webApiProjectStore should be webAPIProjectStore (stylecheck)
pkg/app/server/grpcapi/web_api.go:97:6: ST1003: type webApiEventStore should be webAPIEventStore (stylecheck)
pkg/app/server/grpcapi/web_api.go:101:6: ST1003: type webApiAPIKeyStore should be webAPIAPIKeyStore (stylecheck)
pkg/cli/app.go:33:5: ST1012: error var FlagParseErr should have name of the form ErrFoo (stylecheck)
pkg/config/analysis_template.go:26:2: ST1003: struct field HTTPs should be HTTPS (stylecheck)
pkg/config/application.go:219:2: ST1003: struct field Id should be ID (stylecheck)
pkg/config/application.go:255:2: ST1003: struct field Id should be ID (stylecheck)
pkg/config/application.go:443:2: ST1003: struct field Https should be HTTPS (stylecheck)
pkg/config/control_plane.go:55:2: ST1003: struct field Id should be ID (stylecheck)
pkg/model/application.go:92:27: ST1016: methods on the same type should have the same receiver name (seen 1x "k", 1x "x") (stylecheck)
```

</details>

<details><summary> List of remaining errors </summary>

- [x] stylecheck
- [ ] gocritic
- [ ] errcheck
- [ ] goerr113
- [ ] typecheck
- [ ] gosec
- [x] ~~deadcode, ineffassign~~ https://github.com/pipe-cd/pipecd/pull/4624
- [x] ~~misspell, depguard, unconvert~~ https://github.com/pipe-cd/pipecd/pull/4621
- [x] ~~goimports, gosimple, staticcheck~~ https://github.com/pipe-cd/pipecd/pull/4622

</details>

**Which issue(s) this PR fixes**:

Part of https://github.com/pipe-cd/pipecd/issues/4566

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: pkg/model/application.go [hunk 1]
```diff
@@ -89,8 +89,8 @@ func (a *Application) SetUpdatedAt(t int64) {
 	a.UpdatedAt = t
 }
 
-func (k *ApplicationKind) CompatiblePlatformProviderType() PlatformProviderType {
-	switch *k {
+func (x *ApplicationKind) CompatiblePlatformProviderType() PlatformProviderType {
```
Comment (khanhtc1202) [inline pkg/model/application.go:92]: Why do we need to update this?

**Review by khanhtc1202 — COMMENTED** [review]
Nice, just left a nits 🙌

File: pkg/model/application.go (hunk 1 above)
Comment (karamaru-alpha) [reply pkg/model/application.go]: To resolve this error.

> pkg/model/application.go:92:27: ST1016: methods on the same type should have the same receiver name (seen 1x "k", 1x "x") (stylecheck)

By the way, I reconsidered the naming "k" is more appropriate naming "x", so I unify the naming with A and use a value receiver for the enum by following other implementations.

https://github.com/pipe-cd/pipecd/pull/4638/commits/e8103700923e410bacbceff6f836b64c87469038

Comment (khanhtc1202) [reply pkg/model/application.go]: Got it, thanks 👍

Comment (karamaru-alpha) [reply pkg/model/application.go]: I see. Fixed it👍


https://github.com/pipe-cd/pipecd/pull/4638/commits/5bb575b50aad98e9d0a34d5b43654a1bb8a904ed

**Review by khanhtc1202 — APPROVED** [review]
Thanks 🥇

**Pull request merged**